src/render.rs

```diff
@@ -341,9 +378,7 @@ impl<'a> Context<'a> {
 
         /* Coup */
 
-        self.swapchain = swapchain;
         self.command_fences = command_fences;
-        self.render_pass = render_pass;
```

Comment:
It looks like you've made some changes to the base `render.rs` along with the new text code. If you could split up any changes to a separate commit, that would be great.